boards/common/stm32/include/f0/cfg_clock_default.h

```diff
@@ -31,33 +31,95 @@ extern "C" {
  * @name    Clock settings
  * @{
  */
-/* give the target core clock (HCLK) frequency [in Hz],
- * maximum: 48MHz */
-#define CLOCK_CORECLOCK         MHZ(48)
-/* 0: no external high speed crystal available
- * else: actual crystal frequency [in Hz] */
+/* Select the desired system clock source between PLL, HSE or HSI */
+#ifndef CONFIG_USE_CLOCK_PLL
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || IS_ACTIVE(CONFIG_USE_CLOCK_HSI)
+#define CONFIG_USE_CLOCK_PLL            0
+#else
+#define CONFIG_USE_CLOCK_PLL            1     /* Use PLL by default */
+#endif
+#endif /* CONFIG_USE_CLOCK_PLL */
+
+#ifndef CONFIG_USE_CLOCK_HSE
+#define CONFIG_USE_CLOCK_HSE            0
+#endif /* CONFIG_USE_CLOCK_HSE */
+
+#ifndef CONFIG_USE_CLOCK_HSI
+#define CONFIG_USE_CLOCK_HSI            0
+#endif /* CONFIG_USE_CLOCK_HSI */
+
+#if IS_ACTIVE(CONFIG_USE_CLOCK_PLL) && \
+    (IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || IS_ACTIVE(CONFIG_USE_CLOCK_HSI))
+#error "Cannot use PLL as clock source with other clock configurations"
+#endif
+
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSE) && \
+    (IS_ACTIVE(CONFIG_USE_CLOCK_PLL) || IS_ACTIVE(CONFIG_USE_CLOCK_HSI))
+#error "Cannot use HSE as clock source with other clock configurations"
+#endif
+
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSI) && \
+    (IS_ACTIVE(CONFIG_USE_CLOCK_PLL) || IS_ACTIVE(CONFIG_USE_CLOCK_HSE))
+#error "Cannot use HSI as clock source with other clock configurations"
+#endif
+
+#ifndef CONFIG_BOARD_HAS_HSE
+#define CONFIG_BOARD_HAS_HSE            0
+#endif
+
 #ifndef CLOCK_HSE
-#define CLOCK_HSE               MHZ(8)
+#define CLOCK_HSE                       MHZ(8)
 #endif
-/* 0: no external low speed crystal available,
- * 1: external crystal available (always 32.768kHz) */
-#ifndef CLOCK_LSE
-#define CLOCK_LSE               (0)
+#if IS_ACTIVE(CONFIG_BOARD_HAS_HSE) && (CLOCK_HSE < MHZ(4) || CLOCK_HSE > MHZ(32))
+#error "HSE clock frequency must be between 4MHz and 32MHz"
 #endif
-/* peripheral clock setup */
-#define CLOCK_AHB_DIV           RCC_CFGR_HPRE_DIV1
-#define CLOCK_AHB               (CLOCK_CORECLOCK / 1)
-#define CLOCK_APB1_DIV          RCC_CFGR_PPRE_DIV1      /* max 48MHz */
-#define CLOCK_APB1              (CLOCK_CORECLOCK / 1)
-#define CLOCK_APB2              (CLOCK_APB1)
 
-/* PLL factors */
-#ifndef CLOCK_PLL_PREDIV
-#define CLOCK_PLL_PREDIV        (1)
+#ifndef CONFIG_BOARD_HAS_LSE
+#define CONFIG_BOARD_HAS_LSE            0
+#endif
+#if IS_ACTIVE(CONFIG_BOARD_HAS_LSE)
+#define CLOCK_LSE                       (1)
+#else
+#define CLOCK_LSE                       (0)
```

Comment:
Do you need this definition still then?

Reply:
Yes because CLOCK_LSE is used in the rtc peripheral implementation. And other STM32 families are still using it. Once all this refactoring will be merged we'll be able to get rid of it (only use `CONFIG_BOARD_HAS_LSE`) and/or eventually to assign actual frequency of the LSE clock (32kHz in general but I saw that it could be set to 1MHz on G4).

Reply:
Can you add this to a tracking issue?